### PR TITLE
Add reactor module SQA documents

### PR DESCRIPTION
**Pull request by GiudGiud**
@eshemon @miaoyinb @yjung-anl 
if you have time to review

Most of the documents are templated, the non-templated parts dont need to be too extensive / long, just descriptive enough of what is being done.

### CODE REVIEW

**Review by eshemon — COMMENTED** [review]
Thanks for the heads up, @GiudGiud. This looks fine to me!

File: modules/reactor/doc/content/modules/reactor/sqa/reactor_srs.md [hunk 1]
```diff
@@ -1 +1,47 @@
...
+being so. This includes: liquid metal fast reactors, prismatic high temperature reactors, some variations of
+micro reactors and pressurized water reactors.
+Every design variation of each advanced reactor type cannot be covered by the module.
+It covers the meshing and rotation of control drum in certain reactors.
+
+The system contains, generally, a base set of mesh generators and meshing utilities.
+
+Reactor module developers work with framework and other module and application developers to
+ensure that the Reactor module provides adequate capability to support ongoing and prospective
+research opportunities involving advanced reactors.
```
Comment (cticenhour) [inline modules/reactor/doc/content/modules/reactor/sqa/reactor_srs.md:32]: Just combine these into a single paragraph.

**Review by cticenhour — CHANGES_REQUESTED** [review]
A couple comments to address. Looks good overall!

File: modules/reactor/doc/content/modules/reactor/sqa/reactor_sdd.md [hunk 2]
```diff
@@ -1 +1,64 @@
...
+objects has limited extendability and exists for every simulation configuration that the module is
+capable of running.
+
+!syntax complete subsystems=False actions=False objects=False groups=ReactorApp
+
+!style halign=left
+The MooseApp is the top-level object used to hold all of the other objects in a simulation. In a
+normal simulation a single MooseApp object is created and "run()". This object uses its Factory
+objects to build user defined objects which are stored in a series of Warehouse objects and
+executed. The Finite Element data is stored in the Systems and Assembly object while the domain
```
[the remark this answers is not in the document]
Comment (GiudGiud) [reply modules/reactor/doc/content/modules/reactor/sqa/reactor_sdd.md:54]: finite volume data is actually stored on the mesh (the face infos) not really as much in the assembly

I think in the future we may want to not dwell on any numerical method in particular, there are too many in MOOSE: FE, FV and one day maybe MOC/MC through ray tracing, PIC too

File: modules/reactor/doc/content/modules/reactor/sqa/reactor_srs.md (hunk 1 above)
Comment (GiudGiud) [reply modules/reactor/doc/content/modules/reactor/sqa/reactor_srs.md]: fixed & squashed

File: modules/reactor/doc/content/modules/reactor/sqa/reactor_sdd.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (cticenhour) [reply modules/reactor/doc/content/modules/reactor/sqa/reactor_sdd.md:54]: I would tend to agree with that. Keeps us from having to remember all the places we need to change the documentation. I am fine leaving this as-is for now.

**Review by cticenhour — APPROVED** [review]

**Comment by miaoyinb** [comment]
Thank you, @GiudGiud. Looks good to me.

**Pull request merged**